### ISSUE TITLE
modality encoder question

### ISSUE THREAD

**Issue by FortuneBush** [ZebangCheng/Emotion-LLaMA#26]
"In practice, to save GPU memory, we do not load all Encoders directly onto the GPU but instead load the extracted features“
Does it mean we don't need modality encoder, we already have the llama input token from the different modality encoder, we don't need to get the token from the modality encoder calculation?
Can you tell me more the information about this sentence?
Hoping for your reply!
Thank you!

**Comment by ZebangCheng** [comment]
Sorry for the confusion. In fact, we still need the modality encoders. During training, we freeze the parameters of the encoders so that they are not updated, which means that when the same audio or image is input to the encoder, it always generates the same features. To save GPU memory, we pre-process the audio and image by passing them through the specific encoders to extract features, and then we save these features locally. 

During training, we only load and use the pre-saved features without needing to load the corresponding encoders into memory. However, during deployment, we do load the encoders into memory to process any unseen data. You can refer to our online demo page or the local demo code for further details.

**Comment by wangxinru1202** [comment]
Hello, I would like to ask the following questions:
1. Does the audio input Hubert-Chinese model recognize or extract features, and if so, what features are they?
2.  transcription_en_all.csv.  In this file are the text subtitles of each video, what role does it play in the training stage of the model?
3.Multimodal prompt template:[INST] < AudioFeature> <VideoFeature> [Task Identifier] Prompt [/INST]
Can you elaborate on this part? Hope to give an example.
4.How many samples are there in the test dataset DFEW, and what kind of processing is required?
Maybe my question is very basic, thank you for the answer.

**Comment by ZebangCheng** [comment]
1. The **Hubert-Chinese model** is capable of extracting acoustic features from audio to infer the emotional state of a speaker. Specifically, the feature dimension of the final layer of Hubert-Chinese is [(1+N) * 1024], where 1 represents the class token and N represents the length of the audio. We use the class token as the audio feature for emotion recognition and emotion understanding tasks.

2. The **transcription_en_all.csv** file contains the text of the dialogue in the video, which corresponds to the text modality in multimodal emotion recognition. By combining this with features from other modalities, we can analyze the speaker's stance and true intent, and subsequently infer their emotional state.

3. Our **multimodal prompt template** is as follows:
   ```
   [INST] <VideoFeature> <AudioFeature> [Task Identifier] Prompt [/INST]
   ```
   Here, **<VideoFeature>** refers to the features extracted by the visual encoder, and **<AudioFeature>** refers to the features extracted by the audio encoder. These features are input as tokens in sequence to the large language model. The **[Task Identifier]** includes “emotion” and “reason.” When the **[Task Identifier]** is “emotion,” Emotion-LLaMA performs emotion classification and outputs only the emotion category. When the **[Task Identifier]** is “reason,” Emotion-LLaMA provides the complete reasoning process for the inferred emotion.

4. The **DFEW dataset** contains a total of 16,372 video samples. We follow prior work by removing samples with low vote counts, resulting in the first dataset split (training set with 9,356 samples and test set with 2,341 samples). You will need to use the HuBERT model to extract audio features, and the pre-trained MAE and VideoMAE models to extract visual features.

Please feel free to ask further questions. To facilitate communication, I would appreciate it if you could provide more detailed descriptions of any subsequent queries.

**Comment by wangxinru1202** [comment]
The tuning process is extended to diverse sources, including MER2023 and DFEW, which feature precisely annotated emotional categories. Notably, we utilized the MERR dataset for pre-training the model and then fine-tuned it on target datasets for evaluation. How did the second phase of fine-tuning for a specific dataset work, and can you describe it in more detail?

**Comment by ZebangCheng** [comment]
Emotion-LLaMA was first pre-trained on the MERR dataset (derived from the MER2023 dataset) and achieved excellent zero-shot performance on DFEW, outperforming other multimodal large language models. However, its performance was still lower compared to models specifically trained on DFEW. Therefore, we continued to fine-tune Emotion-LLaMA on the DFEW training set, leading to a significant improvement in performance, reaching state-of-the-art (SOTA) results. The specific instructions used in this phase were the same as those in the previous stage.
